### PR TITLE
Automatically create vnic components if we get metrics that appear to be from a vnic on a known instance.

### PR DESCRIPTION
**Pull request by joshw**

### CODE REVIEW

**Comment by joshw** [comment]
(bump)

File: ZenPacks/zenoss/OpenStackInfrastructure/zenopenstack.py [hunk 1]
```diff
@@ -174,12 +178,28 @@ def set_config(self, device_id, cfg):
...
+            self.component_id[device_id][datasource.resourceId] = datasource.component
+            self.resource_bytype[device_id][datasource.component_meta_type].add(datasource.resourceId)
+
+    def has_resource(self, device_id, resourceId):
+        return resourceId in self.component_id.get(device_id, {})
+
+    def get_component_id(self, device_id, resourceId):
+        # Given a resource ID, return the corresponding component ID,
+        # or None if it is unrecognized.
+        return self.component_id.get(device_id, {}).get(resourceId, None)
```
Comment (pcarinhas) [inline ZenPacks/zenoss/OpenStackInfrastructure/zenopenstack.py]: None, is the default. not needed.

File: ZenPacks/zenoss/OpenStackInfrastructure/zenopenstack.py [hunk 2]
```diff
@@ -897,6 +927,79 @@ def doTask(self):
             self.state = TaskStates.STATE_IDLE
 
 
+class OpenStackVnicTask(BaseTask):
```
Comment (pcarinhas) [inline ZenPacks/zenoss/OpenStackInfrastructure/zenopenstack.py]: Just a comment: seems like we're putting a lot of extraneous code in zenopenstack.

File: ZenPacks/zenoss/OpenStackInfrastructure/zenopenstack.py (hunk 1 above)
Comment (joshw) [reply ZenPacks/zenoss/OpenStackInfrastructure/zenopenstack.py]: True.  I have a habit of doing that from time to time even when i know it's not needed.  I guess I just like the explicit return value.

File: ZenPacks/zenoss/OpenStackInfrastructure/zenopenstack.py (hunk 2 above)
Comment (joshw) [reply ZenPacks/zenoss/OpenStackInfrastructure/zenopenstack.py]: What do you mean?

Comment (pcarinhas) [reply ZenPacks/zenoss/OpenStackInfrastructure/zenopenstack.py]: I think of the services like zenpython and zenopenstack as generic collection services. It just seems like there is a lot of model-specific code going into zenopenstack.

**Pull request merged**